src/storage/local_storage.cpp

```diff
@@ -17,23 +17,36 @@ namespace duckdb {
 LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &table)
     : table_ref(table), allocator(Allocator::Get(table.db)), deleted_rows(0), optimistic_writer(table),
       merged_storage(false) {
+
 	auto types = table.GetTypes();
 	auto data_table_info = table.GetDataTableInfo();
 	auto &io_manager = TableIOManager::Get(table);
 	row_groups = make_shared_ptr<RowGroupCollection>(data_table_info, io_manager, types, MAX_ROW_ID, 0);
 	row_groups->InitializeEmpty();
 
 	data_table_info->GetIndexes().BindAndScan<ART>(context, *data_table_info, [&](ART &art) {
-		if (art.GetConstraintType() != IndexConstraintType::NONE) {
-			// unique index: create a local ART index that maintains the same unique constraint
-			vector<unique_ptr<Expression>> unbound_expressions;
-			unbound_expressions.reserve(art.unbound_expressions.size());
-			for (auto &expr : art.unbound_expressions) {
-				unbound_expressions.push_back(expr->Copy());
-			}
-			indexes.AddIndex(make_uniq<ART>(art.GetIndexName(), art.GetConstraintType(), art.GetColumnIds(),
-			                                art.table_io_manager, std::move(unbound_expressions), art.db));
+		auto constraint_type = art.GetConstraintType();
+		if (constraint_type == IndexConstraintType::NONE) {
+			return false;
+		}
+
+		// UNIQUE constraint.
+		vector<unique_ptr<Expression>> expressions;
+		vector<unique_ptr<Expression>> delete_expressions;
+		for (auto &expr : art.unbound_expressions) {
+			expressions.push_back(expr->Copy());
+			delete_expressions.push_back(expr->Copy());
 		}
+
+		// Create a delete index and a local index.
+		auto delete_index = make_uniq<ART>(art.GetIndexName(), constraint_type, art.GetColumnIds(),
+		                                   art.table_io_manager, std::move(delete_expressions), art.db);
+		delete_index->append_mode = ARTAppendMode::IGNORE_DUPLICATES;
```

Comment:
Maybe we can add a test for a query that deletes the same row multiple times, for example a query that triggers this:

```sql
delete from integers using range(10);
```

Or really anything in the `using` clause that duplicates the rows in the main table.